ludwig/utils/strings_utils.py

```diff
@@ -287,6 +292,43 @@ def create_vocabulary(
     return vocab, str2idx, str2freq, max_line_length, pad_idx, padding_symbol, unknown_symbol
 
 
+def create_vocabulary_single_token(
+    data: DataFrame,
+    num_most_frequent: Optional[int] = None,
+    processor: str = PANDAS,
+    unknown_symbol: str = UNKNOWN_SYMBOL,
+):
+    """Computes a vocabulary over the provided data frame.
+
+    This function is used when the data consists of a single token within one example. E.g., categorical features. In
+    the case of a single token, this function removes some unnecessary tasks and is more efficient. If the feature
```

Comment:
"... is more efficient than create_vocabulary() because this skips tokenization and computing the maximum sequence length, which are unnecessary for category features."